lib/CL/devices/cuda/pocl-cuda.c

```diff
@@ -111,19 +111,23 @@ pocl_cuda_init_device_ops (struct pocl_device_ops *ops)
   /* free_event_data */
 }
 
+/* The CUDA device to be initialized next. This is reset when probing,
+ * and incremented every time a device gets initialized.
+ * TODO this is a bit of a hack, which is also shared by HSA. It might
+ * be a good idea to consider extending the init() function to also
+ * get a progressive index, and use that instead */
+static int next_cuda_device = 0;
```

Comment:
Agree with the comment - this is bit hacky and it would be better if the index was passed to `init()`. I'm fine with this for now though if HSA also does this.